### PR TITLE
Fix password error

### LINKED ISSUE #330: Database error opening messenger 2nd time
**Describe the bug**
Opening the messenger for the 2nd time give me a database error. (it worked before)

**To Reproduce**
- Install messenger
- Open it/Close it and open it for the 2nd time

**Screenshots or Logs**

Database startup error:

Error: SQLITE_NOTADB: file is not a database

**Other information (please complete the following information):**

* Device: Laptop HP Pavilion
* OS: Windows 10
* Loki messenger Version or Git commit hash: 1.0.0-Beta1

### PR DESCRIPTION
**Pull request by BeaudanBrown**
Closes #330 
Latest upstream merge broke the database password setup
Added a simple config check to determine if we should show the password window, and add a field to the database initialise to throw errors if we are trying to unlock with a password

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -746,10 +746,11 @@ app.on('ready', async () => {
 
   // Try to show the main window with the default key
   // If that fails then show the password window
-  try {
-    await showMainWindow(key);
-  } catch (e) {
+  const passwordSet = userConfig.get('passwordSet');
```
Comment (sachaaaaa) [inline main.js]: passwordSet sounds a bit vague?
- dbHasPassword
- databaseIsProtected
- databaseHasPassword
- passwordProtectedDatabase
...

File: password_preload.js [hunk 2]
```diff
@@ -11,6 +11,10 @@ const localeMessages = ipcRenderer.sendSync('locale-data');
 window.theme = config.theme;
 window.i18n = i18n.setup(locale, localeMessages);
 
+window.getEnvironment = () => config.environment;
+window.getVersion = () => config.version;
+window.getAppInstance = () => config.appInstance;
+
```
Comment (sachaaaaa) [inline password_preload.js]: Any reason you moved this up a few lines?

Comment (BeaudanBrown) [reply password_preload.js]: Sorry should have said in description
This was causing the password_preload to fail to load, I assume from the last upstream merge that signal requires these globals available before you require it

File: main.js (hunk 1 above)
Comment (BeaudanBrown) [reply main.js]: True, will update

File: password_preload.js (hunk 2 above)
Comment (sachaaaaa) [reply password_preload.js]: Good catch! Thanks

**Pull request merged**